solutions/dmoj-phidias.mdx

```diff
@@ -0,0 +1,104 @@
+---
+id: dmoj-phidias
+source: DMOJ
+title: phidias
+author: Andrew Cheng
+---
+
+## Solution
+
+Let $f[i][j]$ be the amount of waste material for a rectangle with width $i$ and height $j$. This rectangle of width $i$ and height $j$ can be split into two smaller rectangles (marked in red and green) after removing a plate that Phidias wants (marked in blue). There are two different possible ways to split the remaining shape into two rectangles.
+
+Case1: 
+
+<center>
+
+<table>
+    <tbody>
+        <tr>
+            <td className="bg-red-100 dark:bg-red-800">  1  </td>
+            <td className="bg-red-100 dark:bg-red-800">  1  </td>
+        </tr>
+        <tr>
+            <td className="bg-green-100 dark:bg-green-800">  2  </td>
+            <td className="bg-blue-100 dark:bg-blue-800">  3  </td>
+        </tr>
+    </tbody>
+</table>
+
+</center>
+
+Case2: 
+
+<center>
+
+<table>
+    <tbody>
+        <tr>
+            <td className="bg-red-100 dark:bg-red-800">  1  </td>
+            <td className="bg-green-100 dark:bg-green-800">  2  </td>
+        </tr>
+        <tr>
+            <td className="bg-red-100 dark:bg-red-800">  1  </td>
+            <td className="bg-blue-100 dark:bg-blue-800">  3  </td>
+        </tr>
+    </tbody>
+</table>
+
+</center>
+
+We loop through all $N$ possible marble plates for each $i$ and $j$. Mathematically, the transition formula is as follows:
+
+$f[i][j] = min_k (f[i][j], f[i][j-k.h] + f[i-k.w][k.h], f[i-k.w][j] + f[k.w][j-k.h])$
+
+where $k$ is current marble plate under consideration.
+
+**Time Complexity:**
+$\mathcal{O}(NWH)$
+
+## Implementation
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+
+#include <bits/stdc++.h>
+#define pi pair<int,int>
+#define fi first
+#define se second
+using namespace std;
+
+const int maxW = 605;
+const int maxN = 205;
```

Comment:
`MAX_W` and `MAX_N` instead
also 600 and 200 should work, no? can you change it to that?

Reply:
umm W and H can reach 600 and the same applies for N and 200. So I don't think an array size of exactly 600 / 200 would work.

Reply:
why not?

Reply:
I start my arrays from 1 not 0. I mean I don't think using a couple more bytes of memory would be a big problem.

Reply:
then just still set them to 600 and 200, and put `MAX_W + 1` in your array declarations